### PR TITLE
🐛 Dask gateway: Fix raised exception when service or task is not found

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
fix an issue where dask gateway function would raise an exception when a task is not found

### CODE REVIEW

File: services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py [hunk 1]
```diff
@@ -364,10 +364,21 @@ async def get_next_empty_node_hostname(
...
+            continue
         for task in service_tasks:
             if task["Status"]["State"] in ("new", "pending"):
-                raise TaskNotAssignedError(f"task {task} is not assigned to a host yet")
+                # some task is not running yet. that is a bit weird
+                if (
+                    service_constraints := task.get("Spec", {})
+                    .get("Placement", {})
+                    .get("Constraints", [])
+                ):
```
Comment (GitHK) [inline services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py:375]: I think you can safely remove this if, service_constraints would be a list here.

File: services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py [hunk 2]
```diff
@@ -364,10 +364,21 @@ async def get_next_empty_node_hostname(
...
+                    if len(service_constraints) > 1:
```
Comment (GitHK) [inline services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py:376]: I don't get why there should be more than one task. And in that case what are you doing?

Comment (sanderegg) [reply services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py:376]: these are constraints not tasks. and there can be any number of constraints.

[the remark this answers is not in the document]
Comment (sanderegg) [reply services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py]: well I just gave it a try. let me know if you like it

**Pull request merged**